theano/gpuarray/opt.py

```diff
@@ -943,20 +1302,32 @@ def local_inplace_sparseblockouter(node):
 
 
 # This deals with any abstract convs that have a transfer somewhere
-@register_opt('fast_compile')
+@register_opt('fast_compile', 'conv_dnn')
 @op_lifter([AbstractConv2d,
             AbstractConv2d_gradWeights,
             AbstractConv2d_gradInputs])
-def local_lift_abstractconv2d(node, context_name):
-    if isinstance(node.outputs[0].type, GpuArrayType):
+def local_lift_abstractconv2d(op, context_name, inputs, outputs):
+    if isinstance(outputs[0].type, GpuArrayType):
         # Don't handle this node here, it's already on the GPU.
         return
-    inps = list(node.inputs)
-    inps[0] = as_gpuarray_variable(node.inputs[0],
+    inps = list(inputs)
+    inps[0] = as_gpuarray_variable(inputs[0],
+                                   context_name=context_name)
+    inps[1] = as_gpuarray_variable(inputs[1],
+                                   context_name=context_name)
+    return [op(*inps)]
```

Comment:
You should avoid duplicating code. Can you just call `local_lift_agstractconv2d_graph(...)` here instead?